infrastructure/modules/keyvault/main.tf

```diff
@@ -68,4 +68,8 @@ resource "azurerm_key_vault" "secrets_vault" {
       storage_permissions = []
     }
   }
+
+  lifecycle {
+    prevent_destroy = true
+  }
```

Comment:
This should have been in place from the start - prevents it from being destroyed.